File: backend/db/supabase_client.py
"""
Supabase Client Configuration

This module creates a single, reusable Supabase client for the backend.

IMPORTANT:
- Uses SUPABASE_SERVICE_ROLE_KEY (not anon key) — gives the backend full DB access.
- This key must NEVER be exposed to the frontend.
- Import `supabase` from this module wherever DB access is needed.

Usage:
    from db.supabase_client import supabase

    result = supabase.table("sales").select("*").eq("business_id", biz_id).execute()
"""

import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()

SUPABASE_URL: str = os.environ.get("SUPABASE_URL", "")
SUPABASE_SERVICE_ROLE_KEY: str = (
    os.environ.get("SUPABASE_SERVICE_ROLE_KEY", "")
    or os.environ.get("SUPABASE_ANON_KEY", "")
)

if (
    not SUPABASE_URL
    or not SUPABASE_SERVICE_ROLE_KEY
    or SUPABASE_SERVICE_ROLE_KEY in ("your-service-role-key-here", "your-anon-key-here")
):
    raise EnvironmentError(
        "Missing required environment variables: SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY. "
        "Copy backend/.env.example to backend/.env and fill in your Supabase credentials."
    )

# Single shared client instance (thread-safe for read operations)
supabase: Client = create_client(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY)

File: supabaseClient.ts
/**
 * Supabase Browser Client
 *
 * Used ONLY for authentication on the frontend (login, signup, session management).
 * The frontend NEVER directly queries Supabase tables — all data goes through FastAPI.
 *
 * Architecture:
 *   Frontend → FastAPI → Supabase (data)
 *   Frontend ↔ Supabase Auth (session/JWT only)
 *
 * The Supabase anon key is safe to expose in the browser.
 * The service role key lives ONLY in the backend .env and is never sent here.
 */

import { createClient } from "@supabase/supabase-js";

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!;

if (!supabaseUrl || !supabaseAnonKey) {
  throw new Error(
    "Missing Supabase environment variables. " +
      "Copy frontend/.env.local.example to frontend/.env.local and fill in your credentials."
  );
}

export const supabase = createClient(supabaseUrl, supabaseAnonKey);
